### PR TITLE
Fix several -Wall -Wextra warnings

### PR DESCRIPTION
**Pull request by umar456**

### CODE REVIEW

File: src/api/unified/lapack.cpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ af_err af_pinverse(af_array *out, const af_array in, const double tol,
                    const af_mat_prop options)
 {
     CHECK_ARRAYS(in);
-    return CALL(out, in, options);
+    return CALL(out, in, tol, options);
```
Comment (9prady9) [inline src/api/unified/lapack.cpp]: that an actual bug.

Comment (umar456) [reply src/api/unified/lapack.cpp]: Yeah, we need stronger types internally. Should I submit this separately?

File: src/backend/common/Logger.cpp [hunk 2]
```diff
@@ -51,10 +51,11 @@ loggerFactory(string name) {
 
 string bytesToString(size_t bytes) {
     static array<const char *, 5> units{"B", "KB", "MB", "GB", "TB"};
-    int count = 0;
+    unsigned long long count = 0;
```
Comment (pavanky) [inline src/backend/common/Logger.cpp]: Why not size_t ?

File: src/backend/common/Logger.cpp [hunk 3]
```diff
@@ -51,10 +51,11 @@ loggerFactory(string name) {
...
     double fbytes = static_cast<double>(bytes);
-    for(count = 0; count < units.size() && fbytes > 1000.0; count++) {
-        fbytes *= (1.0 / 1024.0);
+    unsigned long long num_units = units.size();
```
Comment (pavanky) [inline src/backend/common/Logger.cpp]: Same as above.

File: src/api/unified/lapack.cpp (hunk 1 above)
Comment (9prady9) [reply src/api/unified/lapack.cpp]: A separate commit should be fine, I am fine with it in this PR.

File: src/backend/common/Logger.cpp (hunk 2 above)
Comment (umar456) [reply src/backend/common/Logger.cpp]: Yeah, size_t would be better. I am not sure what I was thinking. Fixed.

**Comment by 9prady9** [comment]
@umar456 There seems to be some compilation errors.

**Comment by umar456** [comment]
Addressed all feedback

**Comment by 9prady9** [comment]
@umar456 Why didn't the CI kick in ?

**Pull request merged**